### ISSUE TITLE
The detailerDebug (Seg) Breaks When using comfyui backend on StableSwarmUI

### ISSUE THREAD

**Issue by Maki9009** [ltdrdata/ComfyUI-Impact-Pack#114]
This is the error i get, once it gets to that node it breaks 

Error occurred when executing DetailerForEachDebug:

mat1 and mat2 shapes cannot be multiplied (154x1280 and 2048x640)

File "E:\Comfy\ComfyUI\execution.py", line 151, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "E:\Comfy\ComfyUI\execution.py", line 81, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "E:\Comfy\ComfyUI\execution.py", line 74, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "E:\Comfy\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\impact_pack.py", line 1623, in doit
DetailerForEach.do_detail(image, segs, model, clip, vae, guide_size, guide_size_for, max_size, seed, steps,
File "E:\Comfy\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\impact_pack.py", line 590, in do_detail
enhanced_pil = core.enhance_detail(cropped_image, model, clip, vae, guide_size, guide_size_for_bbox, max_size,
File "E:\Comfy\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\core.py", line 182, in enhance_detail
refined_latent = nodes.KSampler().sample(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise)[0]
File "E:\Comfy\ComfyUI\nodes.py", line 1206, in sample
return common_ksampler(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise)
File "E:\Comfy\ComfyUI\nodes.py", line 1176, in common_ksampler
samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
File "E:\Comfy\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\hacky.py", line 22, in informative_sample
raise e
File "E:\Comfy\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\hacky.py", line 9, in informative_sample
return original_sample(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\sample.py", line 93, in sample
samples = sampler.sample(noise, positive_copy, negative_copy, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
File "E:\Comfy\ComfyUI\comfy\samplers.py", line 693, in sample
samples, _ = sampler.sample_custom(ddim_timesteps=timesteps,
File "E:\Comfy\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\ldm\models\diffusion\ddim.py", line 89, in sample_custom
samples, intermediates = self.ddim_sampling(conditioning, x_T.shape,
File "E:\Comfy\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\ldm\models\diffusion\ddim.py", line 231, in ddim_sampling
outs = self.p_sample_ddim(img, cond, ts, index=index, use_original_steps=ddim_use_original_steps,
File "E:\Comfy\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\ldm\models\diffusion\ddim.py", line 265, in p_sample_ddim
model_output = denoise_function(x, t, **extra_args)
File "E:\Comfy\ComfyUI\comfy\k_diffusion\external.py", line 98, in predict_eps_discrete_timestep
return (input - self(input, sigma, **kwargs)) / utils.append_dims(sigma, input.ndim)
File "E:\Comfy\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\k_diffusion\external.py", line 125, in forward
eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
File "E:\Comfy\ComfyUI\comfy\k_diffusion\external.py", line 151, in get_eps
return self.inner_model.apply_model(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\samplers.py", line 311, in apply_model
out = sampling_function(self.inner_model.apply_model, x, timestep, uncond, cond, cond_scale, cond_concat, model_options=model_options, seed=seed)
File "E:\Comfy\ComfyUI\custom_nodes\ComfyUI_Fooocus_KSampler\sampler\Fooocus\patch.py", line 296, in sampling_function_patched
cond, uncond = calc_cond_uncond_batch(model_function, cond, uncond, x, timestep, max_total_area, cond_concat,
File "E:\Comfy\ComfyUI\custom_nodes\ComfyUI_Fooocus_KSampler\sampler\Fooocus\patch.py", line 266, in calc_cond_uncond_batch
output = model_function(input_x, timestep_, **c).chunk(batch_chunks)
File "E:\Comfy\ComfyUI\comfy\model_base.py", line 61, in apply_model
return self.diffusion_model(xc, t, context=context, y=c_adm, control=control, transformer_options=transformer_options).float()
File "E:\Comfy\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\Comfy\ComfyUI\custom_nodes\ComfyUI_Fooocus_KSampler\sampler\Fooocus\patch.py", line 321, in unet_forward_patched
h = forward_timestep_embed(module, h, emb, context, transformer_options)
File "E:\Comfy\ComfyUI\comfy\ldm\modules\diffusionmodules\openaimodel.py", line 56, in forward_timestep_embed
x = layer(x, context, transformer_options)
File "E:\Comfy\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\ldm\modules\attention.py", line 696, in forward
x = block(x, context=context[i], transformer_options=transformer_options)
File "E:\Comfy\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\ldm\modules\attention.py", line 528, in forward
return checkpoint(self._forward, (x, context, transformer_options), self.parameters(), self.checkpoint)
File "E:\Comfy\ComfyUI\comfy\ldm\modules\diffusionmodules\util.py", line 123, in checkpoint
return func(*inputs)
File "E:\Comfy\ComfyUI\comfy\ldm\modules\attention.py", line 628, in _forward
n = self.attn2(n, context=context_attn2, value=value_attn2)
File "E:\Comfy\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\ldm\modules\attention.py", line 423, in forward
k = self.to_k(context)
File "E:\Comfy\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\Comfy\ComfyUI\comfy\ops.py", line 18, in forward
return torch.nn.functional.linear(input, self.weight, self.bias)

**Comment by ltdrdata** [comment]
This phenomenon occurs when incompatible checkpoints are mixed and used between clips/models.

**Comment by Maki9009** [comment]
soo how does one fix this? because it works normally on comfy, just when using the stableswarmui it breaks

**Comment by ltdrdata** [comment]
> soo how does one fix this? because it works normally on comfy, just when using the stableswarmui it breaks

I cannot provide guidance on SwarmUI as I am not familiar with it. However, I suspect that there might have been some mistakes in configuring the model on the SwarmUI side, rather than anything specific to it.

**Comment by mcmonkey4eva** [comment]
(For the record this is extremely unlikely to relate to swarm itself, more likely just models mixed up or something)

**Comment by ltdrdata** [comment]
> (For the record this is extremely unlikely to relate to swarm itself, more likely just models mixed up or something)

I think so too. However, it seems difficult to provide guidance on whether the workflow was executed from the ComfyUI tab or externally from the Swarm UI, and what settings were used for its execution, as it's unclear.
